src/components/Balance.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { formatEther, parseEther } from 'viem';
+import type { Address } from 'viem';
+import { publicClient } from '../network/client';
+import { useOnchainActionWithCache } from '../hooks/useOnchainActionWithCache';
+
+type BalanceProps = {
+  address: Address;
+  className?: string;
+  decimalDigits?: number;
+  props?: React.HTMLAttributes<HTMLSpanElement>;
+};
+
+const balanceAction = (address: Address) => async (): Promise<string> => {
+  try {
+    return await publicClient
+      .getBalance({
+        address,
+      })
+      .toString();
+  } catch (err) {
+    return '0';
+  }
+};
+
+/**
+ * Balance is a React component that renders the account balance for the given Ethereum address.
+ * It displays the user's balance, with the no. of decimals specified by the 'decimalDigits' prop.
+ * By default, 'decimalDigits' is set to 3.
+ *
+ * @param {Address} props.address - The Ethereum address for which to display the balance.
+ * @param {string} [className] - Optional CSS class for custom styling.
+ * @param {number} [decimalDigits=3] - Determines the no. of decimal digits to be displayed.
+ * @param {React.HTMLAttributes<HTMLSpanElement>} [props] - Additional HTML attributes for the span element.
+ */
+export function Balance({ address, className, decimalDigits = 3, ...props }: BalanceProps) {
```

Comment:
Can we add this functionality to the avatar is possible?

Reply:
not sure if that's the best approach here - generally, it's best to keep each component small and atomic, with a single, clear responsibility

adding balance data to the avatar might add multiple responsibilities inside a single component

if users need balance data in their avatar, having a separate component allows them to stitch the avatar and balance components together in the way they feel fits their use case best

LMK your thoughts on this

Reply:
This week is hard for us to ship a new component.

Let's table this till the end of ETH Denver.

But, I love we at least started the convo.